app/autocomplete_mention/routes.test.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+var expect = require('chai').expect
+  , request = require('supertest');
+
+var app = require('../../app');
+
+describe('Autocomplete', function() {
+  describe('GET /mention', function() {
+    it('should send a response', function(done) {
+      request(app)
+        .get('/autocomplete/mention')
+        .expect(200)
+        .end(function(err,res) {
+          if(err) return done(err);
+          expect(res).to.exist;
+          done();
+        });
+    });
+
+    it('should respond with json', function(done) {
+      request(app)
+        .get('/autocomplete/mention?pattern=wor')
+        .set('Accept','application/json')
+        .expect('Content-Type',/json/)
+        .expect(200,done);
+    });
+
+    var users = [
+      { name : 'world' },
+      { name : 'work' },
+      { name : 'woman' },
+    ];
+
+    it('should respond with matches', function(done) {
+      require('../../lib/mongo').connect(function(err,db) {
+        db.collection('user').insert( users ,
+          function(err,result) {
+            if(err) throw err;
+            request(app)   // seeking matches for wor
+              .get('/autocomplete/mention?pattern=wor')
+              .expect(200)
+              .end(function(err, res) {
+                if(err) return done(err);
+
+                expect(res.text).to.contain('world');
```

Comment:
It's preferred to check the exact response here since you should know it.

`expect(res.body).to.deep.equal([{name: 'world'}, {name: 'work'}]);`